Automattic/wp-calypso
# Replace build-packages:watch to only rebuild changed package

#### Changes proposed in this Pull Request

The `build-packages:watch` npm script was added in #37802 to allow rebuilding the `@automattic/composite-checkout` package when it changed in the monorepo (and to also trigger a rebuild of calypso so that the modifications would be used). However, it causes a rebuild of _all_ packages any time a file inside a single package has changed. This can be a very slow process.

In this PR we introduce a new script that allows rebuilding only the packages which contained the changed file. This should be considerably faster. (In my tests it went from about 120 seconds down to about 10 seconds depending on the package changed).

This also ignores changes to Markdown files and files in test directories in `packages/`.

#### Testing instructions

- Run `yarn start` and wait for calypso to be started.
- Modify a source file inside the `packages` directory.
- Verify that you see a `rebuidling` message for the package which contains the changed file. 
- Verify that the change you made took effect by loading http://calypso.localhost:3000 and looking for your change. A good change to make is a `console.log()` statement for which it is easy to see the results.
